domain/machine/state/machine_cloud_instance_test.go

```diff
@@ -27,7 +27,47 @@ func (s *stateSuite) TestGetHardwareCharacteristics(c *gc.C) {
 	c.Check(*hc.RootDiskSource, gc.Equals, "/test")
 	c.Check(*hc.CpuCores, gc.Equals, uint64(4))
 	c.Check(*hc.CpuPower, gc.Equals, uint64(75))
-	c.Check(*hc.AvailabilityZone, gc.Equals, "deadbeef-0bad-400d-8000-4b1d0d06f00d")
+	c.Check(*hc.AvailabilityZone, gc.Equals, "az-1")
+	c.Check(*hc.VirtType, gc.Equals, "virtual-machine")
+}
+
+func (s *stateSuite) TestGetHardwareCharacteristicsWithoutAvailabilityZone(c *gc.C) {
+	db := s.DB()
+	// Create a reference machine.
+	err := s.state.CreateMachine(context.Background(), "42", "", "")
+	c.Assert(err, jc.ErrorIsNil)
+	var machineUUID string
+	row := db.QueryRowContext(context.Background(), "SELECT uuid FROM machine WHERE name='"+string("42")+"'")
+	c.Assert(row.Err(), jc.ErrorIsNil)
+	err = row.Scan(&machineUUID)
```

Comment:
**suggestion:** Maybe we can do it in one line ?

```go
err := db.QueryContext(/*...*/).Scan(&machineUUID)
c.Assert(err, jc.ErrorIsNil)
```

Reply:
Sure, done.